### DISCUSSION TITLE
adjusting ios-src on selection from a slider.

### DISCUSSION THREAD

**Discussion by Baser-T** [google/model-viewer#2082] [Q&A]
Hey there,

I'm currently in the process of implementing the slider that is highlighted on your website. in the code the following line of code is used to change to source of the 3d object:
`  <script type="module">
    const modelViewer = document.querySelector("model-viewer");

    window.switchSrc = (element, name) => {
      const base = "assets/" + name;
      modelViewer.src = base + '.gltf';
      modelViewer.poster = base + '.png';
      const slides = document.querySelectorAll(".slide");
      slides.forEach((element) => { element.classList.remove("selected"); });
      element.classList.add("selected");
    };

    document.querySelector(".slider").addEventListener('beforexrselect', (ev) => {
      // Keep slider interactions from affecting the XR scene.
      ev.preventDefault();
    });
  </script>`

however I've so far not found a method to also change to .usdz file to implement the IOS compatibility. For example I would expect something like modelViewer.ios-src = base + ".usdz". Did I overlook something or is that functionality not yet in?

Thanks in advance!

**Comment by Baser-T** [comment]
Just wanted to add, I also noticed that when I use the tutorial of the change material base color. While it works flawlessly in the model-viewer frame itself, the color is lost when I switch to AR on my phone. Do I have to pass that color along somewhere or is it not yet available?

**Reply by elalish** [reply]
It should work in WebXR mode, as that's the one that stays in the browser. Since we can only pass a model URL to the SceneViewer and QuickLook apps, there's no way to customize them.

**Reply by Baser-T** [reply]
Alright I see, thanks!

**Comment by elalish** [comment]
You're almost there: JS APIs use camelCase, so it's `modelViewer.iosSrc = base + ".usdz"`.

**Reply by Baser-T** [reply]
Aah, indeed very close. Thanks a lot!
